lib/ansible/modules/cloud/amazon/aws_s3.py

```diff
@@ -427,7 +453,10 @@ def create_bucket(module, s3, bucket, location=None):
 def paginated_list(s3, **pagination_params):
     pg = s3.get_paginator('list_objects_v2')
     for page in pg.paginate(**pagination_params):
-        yield [data['Key'] for data in page.get('Contents', [])]
+        if LIST_EXTENDED == True:
```

Comment:
For example, module is available in list_keys()
You could set a var there, like at L463:
``list_type = module.params.get("mode")``
Then pass it into paginated_list to be checked here and avoid the global entirely.

Reply:
Hi @jillr I decided to follow this approach and pass mode as an argument to paginated_list, removing LIST_EXTENDED all-together. I have also implemented test/integration/targets/aws_s3/tasks/main.yml. Thanks for your input.